webtau-core/src/main/java/com/twosigma/webtau/expectation/equality/handlers/DateAndStringCompareToHandler.java

```diff
@@ -0,0 +1,168 @@
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import static com.twosigma.webtau.utils.TraceUtils.renderValueAndType;
+
+public class DateAndStringCompareToHandler implements CompareToHandler {
+    private static final ZoneId UTC = ZoneId.of("UTC");
+
+    private static final List<FormatParser> parsers = Arrays.asList(
+            new FormatParser(DateTimeFormatter.ISO_DATE_TIME, ZonedDateTime::parse),
+            new FormatParser(DateTimeFormatter.ISO_LOCAL_DATE, LocalDate::parse));
+
+    @Override
+    public boolean handleEquality(Object actual, Object expected) {
+        return handle(actual, expected);
+    }
+
+    @Override
+    public boolean handleGreaterLessEqual(Object actual, Object expected) {
+        return handle(actual, expected);
+    }
+
+    @Override
+    public void compareEqualOnly(CompareToComparator comparator, ActualPath actualPath, Object actual, Object expected) {
+        Comparator dateComparator = new Comparator(comparator, actualPath, actual, expected, true);
+        dateComparator.compare();
+    }
+
+    @Override
+    public void compareGreaterLessEqual(CompareToComparator comparator, ActualPath actualPath, Object actual, Object expected) {
+        Comparator dateComparator = new Comparator(comparator, actualPath, actual, expected, false);
+        dateComparator.compare();
+    }
+
+    private boolean handle(Object actual, Object expected) {
+        return actual instanceof String && (
+                expected instanceof LocalDate ||
+                        expected instanceof ZonedDateTime);
+    }
+
+    private class Comparator {
+        private final CompareToComparator compareToComparator;
+        private final ActualPath actualPath;
+        private final TemporalAccessor actual;
+        private final Object expected;
+        private final boolean isEqualOnly;
+
+        Comparator(CompareToComparator compareToComparator, ActualPath actualPath,
+                   Object actual, Object expected, boolean isEqualOnly) {
+            this.compareToComparator = compareToComparator;
+            this.actualPath = actualPath;
+            this.actual = actualToTemporalAccessor(actual);
+            this.expected = expected;
+            this.isEqualOnly = isEqualOnly;
+        }
+
+        void compare() {
+            if (actual instanceof LocalDate && expected instanceof LocalDate) {
+                compareLocalDates((LocalDate) actual, (LocalDate) expected);
+            } else if (actual instanceof ZonedDateTime && expected instanceof LocalDate) {
+                compareZonedDateTimeAndLocalDate((ZonedDateTime) actual, (LocalDate) expected);
+            } else if (actual instanceof ZonedDateTime && expected instanceof ZonedDateTime) {
+                compareZonedDateTimes((ZonedDateTime) actual, (ZonedDateTime) expected);
+            } else {
+                throw new UnsupportedOperationException("combination is not supported:\n" +
```

Comment:
Why not actual LocalDate with expected ZonedDateTime?

Reply:
got mentally drained by this time :) do you want to give it a shot?
Plus I didn't have a use case to be sufficiently motivated.

Reply:
Hmm, yeah, what time portion it should use is a good question.  Let's park it until we have a usecase.